ts/packages/anchor/src/provider.ts

```diff
@@ -1,50 +1,74 @@
 import {
-  Connection,
-  Signer,
-  PublicKey,
+  Base58EncodedAddress,
+  createDefaultRpcTransport,
+  createSolanaRpc,
+  setTransactionFeePayer,
+  setTransactionLifetimeUsingBlockhash,
+  signTransaction,
   Transaction,
-  TransactionSignature,
-  ConfirmOptions,
-  SimulatedTransactionResponse,
-  Commitment,
-  SendTransactionError,
-  SendOptions,
-  VersionedTransaction,
-  RpcResponseAndContext,
-} from "@solana/web3.js";
+} from "web3js-experimental";
 import { bs58 } from "./utils/bytes/index.js";
-import { isBrowser, isVersionedTransaction } from "./utils/common.js";
-import {
-  simulateTransaction,
-  SuccessfulTxSimulationResponse,
-} from "./utils/rpc.js";
+import { isBrowser } from "./utils/common.js";
+import { SuccessfulTxSimulationResponse } from "./utils/rpc.js";
+
+// Type helpers for non-exported Web3 JS types.
+// As of right now, these types will not be exported from the new API.
```

Comment:
Given types won't have an effect on the bundle size, why are these types not exported?

Reply:
We're not _opposed_ to export types, we just want to do it _very_ thoughtfully, and only when there's no other way. The current web3.js suffers from _extreme_ type exporting, and it makes the library extremely difficult to refactor (since someone, somewhere, is relying on every type).

Reply:
Thanks for clarifying but wouldn't making people derive the types themselves cause problems if/when there is a refactor of internal `web3.js` types anyway?

https://github.com/coral-xyz/anchor/blob/2e2a51b3a02dd88d5c9f205dddfb92819f9555c5/ts/packages/anchor/src/provider.ts#L20-L30

For example, the types defined/derived here such as `Commitment` and the return types of RPC methods are not JS specific and are unlikely to change but if they were to change, it would cause errors in Anchor even though they are not "exported".